kratos/python_scripts/json_output_process.py

```diff
@@ -243,147 +244,147 @@ def ExecuteFinalizeSolutionStep(self):
+                            elif variable_type == "Vector":
                                 if not self.resultant_solution:
-                                    list = self.__kratos_vector_to__python_list(value)
-                                    data[node_identifier][variable_name ].append(list)
+                                    data[node_identifier][variable_name].append(value)
                                 else:
-                                    aux = 0.0
-                                    for index in range(len(value)):
-                                        aux += value[index]
                                     if count == 0:
-                                        data["RESULTANT"][variable_name ].append(aux)
+                                        data["RESULTANT"][variable_name].append(value)
                                     else:
-                                        data["RESULTANT"][variable_name ][-1] += aux
-                        elif variable_type == "Vector":
-                            if not self.resultant_solution:
-                                data[node_identifier][variable_name].append(value)
-                            else:
-                                if count == 0:
-                                    data["RESULTANT"][variable_name].append(value)
-                                else:
-                                    data["RESULTANT"][variable_name][-1] += value
+                                        data["RESULTANT"][variable_name][-1] += value
 
-                        # TODO: Add pending classes
-                    count += 1
+                            # TODO: Add pending classes
+                        count += 1
 
             count = 0
             # Gauss points values
-            for elem in self.sub_model_part.Elements:
-                compute = self.__check_flag(elem)
+            if len(self.gauss_points_output_variables) > 0:
+                for elem in self.sub_model_part.Elements:
+                    compute = self.__check_flag(elem)
 
-                if compute:
-                    for i in range(self.params["gauss_points_output_variables"].size()):
-                        out = self.params["gauss_points_output_variables"][i]
-                        variable_name = out.GetString()
-                        variable = KratosMultiphysics.KratosGlobals.GetVariable(variable_name)
-                        variable_type = KratosMultiphysics.KratosGlobals.GetVariableType(variable_name)
-                        value = elem.CalculateOnIntegrationPoints(variable, self.sub_model_part.ProcessInfo)
+                    if compute:
+                        for i in range(len(self.gauss_points_output_variables)):
+                            variable = self.gauss_points_output_variables[i]
+                            variable_name = variable.Name()
+                            variable_type = KratosMultiphysics.KratosGlobals.GetVariableType(variable_name)
```

Comment:
this is slow, could it be done outside the loop?

Reply:
Maybe

Reply:
Thinking thinking

Reply:
please dont refactor the entire class, it can also be left as is :D

Reply:
I have an indea, for a next PR